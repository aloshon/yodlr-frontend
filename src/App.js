import Routes from './Routes';
import axios from "axios";
import './css/App.css';
import StateContext from "./StateContext";

/**
 * App renders routes component and sets basic theme styles
 */
function App() {
  // updates user state and return edited user
  const updateState = async ({id, firstName, lastName, email, state, isAdmin}) => {

    let user = {
        id,
        email,
        firstName,
        lastName,
        state,
        isAdmin
    }
    
    try{
        const {data} = await axios.put(`http://localhost:3001/users/${id}`,
        user);
        
        return data
    } catch(e){
        alert("Error changing state");
        console.error(e);
    }
  };


  return (
    <StateContext.Provider value ={{updateState}}>
      <div className="App">
        <header className="App-header">
          <Routes />
        </header>
      </div>
    </StateContext.Provider>
  );
}

export default App;
