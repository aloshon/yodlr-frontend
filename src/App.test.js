import { render, screen } from '@testing-library/react';
import App from './App';
import {BrowserRouter} from 'react-router-dom';

describe("app should render register since there is no logged in uesr", () => {
  const {asFragment, getByLabelText, getByText} = render(<BrowserRouter><App /></BrowserRouter>);

  test('renders register page inputs', () => {
      const email = getByLabelText("Email");
      const first = getByLabelText("First Name");
      const last = getByLabelText("Last Name");
      const submit = getByText("Submit");
      expect(email).toBeInTheDocument();
      expect(first).toBeInTheDocument();
      expect(last).toBeInTheDocument();
      expect(submit).toBeInTheDocument();
  });
  
  test("matches snapshot", () => {
    expect(asFragment()).toMatchSnapshot();
  });
})
