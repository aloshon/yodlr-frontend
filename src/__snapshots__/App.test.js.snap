// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app should render register since there is no logged in uesr matches snapshot 1`] = `<DocumentFragment />`;
